core/market/src/negotiation/common.rs

```diff
@@ -692,7 +688,10 @@ impl CommonBroker {
             Owner::Provider => &proposal.negotiation.provider_id != caller_id,
             Owner::Requestor => &proposal.negotiation.requestor_id != caller_id,
         } {
-            ProposalValidationError::Unauthorized(proposal.body.id.clone(), caller_id.clone());
+            return Err(ProposalValidationError::Unauthorized(
```

Comment:
It would be worth to determine (debug) whether it was just a failed attempt to make system secure (so no Node could send Proposal with proposal/requestor ids different than its own), or whether it was supposed to reject majority of Proposals early?

Reply:
Made new issue & left dodo here